### PR TITLE
Allow `title` on <LinkButton>s

### PR DESCRIPTION
**Pull request by j-f1**
This allows adding tooltips to links.

### CODE REVIEW

**Comment by shiftkey** [comment]
@j-f1 can you elaborate a bit on what this enables for you? I think having concerns like `style` over in stylesheets is easier to maintain, but perhaps there's other things I've not thought about here...

**Comment by j-f1** [comment]
@shiftkey There is currently no way to set a `title`.

**Comment by shiftkey** [comment]
> There is currently no way to set a title.

Aha! I'll point to some prior art on this in the [Avatar](
https://github.com/desktop/desktop/blob/3edf895663d66ef816338dc79155e808e3c9070c/app/src/ui/lib/avatar.tsx#L12) component.

I think exposing `title` as a new prop, rather than inheriting and enabling all the props to be set, is a better approach here. Is this related to #1948?

**Comment by j-f1** [comment]
@shiftkey Yes, it’s related.

**Comment by j-f1** [comment]
Why is it a bad idea to allow any prop?

**Comment by shiftkey** [comment]
> Why is it a bad idea to allow any prop?

A couple of reasons off the top of my head:

 - friendly abstractions - we've defined a group of controls for common UI elements under `app/src/ui/lib`, rather than duplicating raw JSX or using the abstractions directly
 - potential for misuse - you mentioned enabling `style` usage, but we'd rather people use stylesheets and not inline styles - this pattern makes the contract between the control and the caller easier to understand
 - more props means more opportunities for react to do rendering when things change - by only exposing a subset that the app needs, we reduce the surface area for unnecessary re-rendering

File: app/src/ui/lib/link-button.tsx [hunk 1]
```diff
@@ -4,7 +4,7 @@ import * as classNames from 'classnames'
 
 interface ILinkButtonProps {
   /** A URI to open on click. */
-  readonly uri?: string
+  readonly uri?: string | JSX.Element | ReadonlyArray<JSX.Element>
```
Comment (shiftkey) [inline app/src/ui/lib/link-button.tsx]: Given this is what the user clicks, it should be just an optional `string`.

Comment (j-f1) [inline app/src/ui/lib/link-button.tsx]: Sorry, meant to add that to `children`.

File: app/src/ui/lib/link-button.tsx [hunk 2]
```diff
@@ -20,22 +20,20 @@ interface ILinkButtonProps {
...
 }
 
 /** A link component. */
 export class LinkButton extends React.Component<ILinkButtonProps, void> {
   public render() {
-    const href = this.props.uri || ''
-    const className = classNames('link-button-component', this.props.className)
+    const { uri, className, ...otherProps } = this.props
+    const href = uri || ''
+    const props = { ...otherProps, className: classNames('link-button-component', className), onClick: this.onClick, href }
```
Comment (j-f1) [inline app/src/ui/lib/link-button.tsx]: @shiftkey Which JSX approach?

Comment (shiftkey) [inline app/src/ui/lib/link-button.tsx]: The explicit props being assigned:

```tsx
<a
  className={className}
  href={href}
  onClick={this.onClick}
  tabIndex={this.props.tabIndex}
  disabled={this.props.disabled}
>
```

Comment (j-f1) [inline app/src/ui/lib/link-button.tsx]: Fixed.

**Comment by shiftkey** [comment]
@j-f1 thanks for this!

**Pull request merged**